controllers/multiclusterobservability/multiclusterobservability_controller.go

```diff
@@ -476,13 +416,28 @@ func (r *MultiClusterObservabilityReconciler) SetupWithManager(mgr ctrl.Manager)
 			return false
 		},
 	}
-	// Watch the Secret
-	err = c.Watch(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestForObject{}, pred)
-	if err != nil {
-		return err
-	}
+
+	// create a new controller and start watch for relevant resources
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&mcov1beta2.MultiClusterObservability{}).
+		// Watch for changes to primary resource MultiClusterObservability with predicate
+		For(&mcov1beta2.MultiClusterObservability{}, builder.WithPredicates(mcoPred)).
+		// Watch for changes to secondary resource Deployment and requeue the owner Observatorium
+		Watches(&source.Kind{Type: &appsv1.Deployment{}}, &handler.EnqueueRequestForObject{}, builder.WithPredicates(deployPred)).
+		// Watch for changes to secondary resource statefulSet and requeue the owner Observatorium
+		Watches(&source.Kind{Type: &appsv1.StatefulSet{}}, &handler.EnqueueRequestForObject{}, builder.WithPredicates(statefulsetPred)).
+		// Watch for changes to secondary resource ConfigMap and requeue the owner MultiClusterObservability
+		Owns(&corev1.ConfigMap{}).
+		// Watch for changes to secondary resource Secret and requeue the owner MultiClusterObservability
+		Owns(&corev1.Secret{}).
+		// Watch for changes to secondary resource Service and requeue the owner MultiClusterObservability
+		Owns(&corev1.Service{}).
```

Comment:
what's the purpose to include the Owns() functions here? Means watch the owned resources for all kinds of events w/o any custom filter? If that, will it introduce conflict?
e.g. 
Owns(&corev1.Secret{}) and Watches(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestForObject{}, builder.WithPredicates(secretPred))

Reply:
@marcolan018 I tried to keep the old logic, but set up the controller with new controller [builder](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.6.1/pkg/builder). 
According to the doc:

1. the old watch with `EnqueueRequestForOwner`
```
c.Watch(&source.Kind{Type: &corev1.Service{}}, &handler.EnqueueRequestForOwner
``` is equal to the `Owns() function`.

If you check the old logic, we also watch twice, once for owner type, once for `EnqueueRequestForObject` type, I have a diccussion with @clyang82 
He told me that the multiple watch/own should not conflict, they will register multiple hook functions, so if one hook function match, it will create reconcile request.